Sources/Sentry/SentryOptions.m

```diff
@@ -230,4 +239,26 @@ - (void)addInAppExclude:(NSString *)inAppExclude
     _inAppExcludes = [self.inAppExcludes arrayByAddingObject:inAppExclude];
 }
 
+- (void)setSampleRate:(NSNumber *)sampleRate
+{
+    if (sampleRate == nil) {
+        _sampleRate = nil;
+    } else if ([sampleRate doubleValue] >= 0 && [sampleRate doubleValue] <= 1.0) {
+        _sampleRate = sampleRate;
+    } else {
+        _sampleRate = _defaultSampleRate;
+    }
+}
+
+- (void)setTracesSampleRate:(NSNumber *)tracesSampleRate
+{
+    if (tracesSampleRate == nil) {
+        _tracesSampleRate = nil;
+    } else if ([tracesSampleRate doubleValue] >= 0 && [tracesSampleRate doubleValue] <= 1.0) {
```

Comment:
L: What about we assign `[tracesSampleRate doubleValue]` and just use that value  on the second hand of `&&`?

Reply:
Done